internal/app/go-filecoin/node/node.go

```diff
@@ -431,12 +430,34 @@ func (node *Node) setupStorageMining(ctx context.Context) error {
 
 	waiter := msg.NewWaiter(node.chain.ChainReader, node.chain.MessageStore, node.Blockstore.Blockstore, node.Blockstore.CborStore)
 
-	sub, err := submodule.NewStorageMiningSubmodule(minerAddr, workerAddr, node.Repo.Datastore(), sectorBuilder, &node.chain, &node.Messaging, waiter, &node.Wallet)
+	node.StorageMining, err = submodule.NewStorageMiningSubmodule(minerAddr, workerAddr, node.Repo.Datastore(), sectorBuilder, &node.chain, &node.Messaging, waiter, &node.Wallet)
 	if err != nil {
 		return err
 	}
 
-	node.StorageMining = &sub
+	node.StorageProtocol, err = submodule.NewStorageProtocolSubmodule(
+		ctx,
+		minerAddr,
+		address.Undef, // TODO: This is for setting up mining, we need to pass the client address in if this is going to be a storage client also
+		&node.chain,
+		&node.Messaging,
+		waiter,
+		node.StorageMining.PieceManager,
+		node.Wallet.Wallet,
+		node.Host(),
+		node.Repo.Datastore(),
+		node.Blockstore.Blockstore,
+		repoPath,
+		node.PorcelainAPI.MinerGetWorkerAddress)
+	if err != nil {
+		return errors.Wrap(err, "error initializing storage protocol")
+	}
+
```

Comment:
If we're writing TODOs here another one is to separate the module creation out of the node setup mining function.  The module should be created in the builder and then started here.

Reply:
TODO added with issue #3738